hack/build-cluster-api.sh

```diff
@@ -9,7 +9,7 @@ fi
 
 TARGET_OS_ARCH=$(go env GOOS)_$(go env GOARCH)
 CLUSTER_API_BIN_DIR="${PWD}/cluster-api/bin/${TARGET_OS_ARCH}"
-CLUSTER_API_MIRROR_DIR="${PWD}/pkg/cluster-api/mirror/"
+CLUSTER_API_MIRROR_DIR="${PWD}/pkg/clusterapi/mirror/"
```

Comment:
This is mirroring what Terraform does today, I'm happy to change if needed although we need to package the zip/binaries using embed.FS

Reply:
For example, today no Terraform binaries or zip are built in `/bin`; given that they're packaged within the binary

Reply:
I don't know if there was any hystorical reason/motivation for the terraform packaging, probably @patrickdillon or @r4f4 may have some additional bits for that. It doesn't want of course to be a blocking point, but since it's a fresh start it may be worth evaluating it.

Reply:
@jhixson74 may have insight into any historical reasons. I think keeping all the binaries in a single /bin would simplify for development, but might complicate ux--users shouldn't be using these binaries directly.